diff -urN gpc++0.5.2_oryg/include/gp.h gpc++0.5.2/include/gp.h
--- gpc++0.5.2_oryg/include/gp.h	1997-02-16 13:51:20.000000000 +0100
+++ gpc++0.5.2/include/gp.h	2005-06-02 14:48:58.000000000 +0200
@@ -49,7 +49,8 @@
 #ifndef __GP_H
 #define __GP_H
 
-#include <iostream.h>
+#include <iostream>
+using namespace std;
 
 
 
@@ -398,7 +399,7 @@
   friend int operator == (GPGene& pg1, GPGene& pg2);
   virtual int compare (GPGene& g);
 
-  friend GP;
+  friend class GP;
 
   GPGene& operator = (GPGene& gpo) {
     GPExitSystem ("operator =", "Assignment operator not yet implemented"); 
@@ -462,7 +463,7 @@
   void resolveNodeValues (GPAdfNodeSet& adfNs);
   virtual GPObject* createObject() { return new GP; }
 
-  friend GPPopulation;
+  friend class GPPopulation;
 
   GP& operator = (GP& gpo) {
     GPExitSystem ("operator =", "Assignment operator not yet implemented"); 
diff -urN gpc++0.5.2_oryg/Makefile gpc++0.5.2/Makefile
--- gpc++0.5.2_oryg/Makefile	1997-02-16 13:51:37.000000000 +0100
+++ gpc++0.5.2/Makefile	2005-06-04 15:57:45.000000000 +0200
@@ -5,39 +5,18 @@
 
 all:
 	cd src; make
-	cd symbreg; make
-	cd lawn; make
-	cd skeleton; make
-	cd ant; make
 
 clean:
 	rm -f *~
 	cd src; make clean
-	cd symbreg; make clean
-	cd lawn; make clean
-	cd skeleton; make clean
-	cd ant; make clean
 
 superclean: 
 	rm -f *~
 	cd src; make superclean
-	cd symbreg; make superclean
-	cd lawn; make superclean
-	cd skeleton; make superclean
-	cd ant; make superclean
 
 # Strip debug information (if there is any). 
 strip:
 	cd src; make strip
-	cd lawn; make strip
-	cd skeleton; make strip
-	cd ant; make strip
-	cd symbreg; make strip
 
 install:
-	$(INSTALL) -d $(INSTALLDIR)/include
-	$(INSTALL) -d $(INSTALLDIR)/lib
-	$(INSTALL) -f $(INSTALLDIR)/lib lib/$(GPLIBFILE)
-	$(INSTALL) -f $(INSTALLDIR)/include include/gp.h
-	$(INSTALL) -f $(INSTALLDIR)/include include/gpconfig.h
-
+	cd src; $(MAKE) install
diff -urN gpc++0.5.2_oryg/Makefile.ini gpc++0.5.2/Makefile.ini
--- gpc++0.5.2_oryg/Makefile.ini	1997-02-16 13:51:37.000000000 +0100
+++ gpc++0.5.2/Makefile.ini	2005-06-04 18:05:41.000000000 +0200
@@ -2,10 +2,10 @@
 # Location of the GP include files and the GP library
 GPINCLUDE	= ../include
 GPLIB		= ../lib
-
+GPFILE		= libgpc++.so
 # Name of GP library.  Change '.a' to '.so' if you want to use dynamic
 # libraries, and change the archive command in src/Makefile as well.
-GPLIBFILE = $(GPLIB)/libgp.a
+GPLIBFILE = $(GPLIB)/$(GPFILE)
 
 # Compiler flags.  Use -O to optimise
 OVERALLCPPFLAGS	= -g
@@ -25,4 +25,6 @@
 
 # Install command and installation directory
 INSTALL		= install
-INSTALLDIR	= /usr/local
+INSTALLDIR	= /usr/lib
+INCLUDEDIR	= /usr/include/gpc++
+MKDIR		= mkdir -p
\ Brak znaku nowej linii na koñcu pliku
diff -urN gpc++0.5.2_oryg/src/config.cc gpc++0.5.2/src/config.cc
--- gpc++0.5.2_oryg/src/config.cc	1997-02-16 13:51:23.000000000 +0100
+++ gpc++0.5.2/src/config.cc	2005-06-03 17:23:57.000000000 +0200
@@ -84,9 +84,9 @@
 Thomas Weinbrenner 1996, 1997
 
 */
-
-#include <fstream.h>
-#include <iomanip.h>
+using namespace std;
+#include <fstream>
+#include <iomanip>
 #include <stdlib.h>
 #include <string.h>
 #include <ctype.h>
diff -urN gpc++0.5.2_oryg/src/gpv.cc gpc++0.5.2/src/gpv.cc
--- gpc++0.5.2_oryg/src/gpv.cc	1997-02-16 13:51:25.000000000 +0100
+++ gpc++0.5.2/src/gpv.cc	2005-06-03 17:25:19.000000000 +0200
@@ -48,7 +48,7 @@
 
 
 #include <stdlib.h>
-#include <fstream.h>
+#include <fstream>
 
 #include "gp.h"
 
diff -urN gpc++0.5.2_oryg/src/Makefile gpc++0.5.2/src/Makefile
--- gpc++0.5.2_oryg/src/Makefile	1997-02-16 13:51:26.000000000 +0100
+++ gpc++0.5.2/src/Makefile	2005-06-04 19:03:30.000000000 +0200
@@ -38,9 +38,9 @@
 # Create the GP library. Use either ld (for dynamic libraries) or ar
 # and ranlib.
 $(TARGET) : $(OBJS) 
-	$(AR) -r $(TARGET) $(OBJS)
-	$(RANLIB) $(TARGET)
-#	$(LD) -r -o $(TARGET) $(OBJS)
+	$(AR) -r lib/gpc++.la $(OBJS)
+#	$(RANLIB) gpc++.la
+	$(LD) -r -o $(TARGET) $(OBJS)
 
 clean:
 	rm -f $(OBJS) *~
@@ -72,3 +72,8 @@
 node.o: node.cc ../include/gp.h
 pop.o: pop.cc ../include/gp.h
 select.o: select.cc ../include/gp.h
+
+install:
+	$(INSTALL) $(GPLIBFILE)	$(INSTALLDIR)
+#	$(INSTALL) ./.libs/libgpc++.so	$(INSTALLDIR)
+#	$(INSTALL) ../include/* $(INCLUDEDIR)
diff -urN gpc++0.5.2_oryg/src/misc.cc gpc++0.5.2/src/misc.cc
--- gpc++0.5.2_oryg/src/misc.cc	1997-02-16 13:51:25.000000000 +0100
+++ gpc++0.5.2/src/misc.cc	2005-06-03 17:25:26.000000000 +0200
@@ -44,7 +44,7 @@
 
 // Simply an exit system, which outputs some error
 
-#include <iostream.h>
+#include <iostream>
 #include <time.h>
 #include <stdlib.h>
 
diff -urN gpc++0.5.2_oryg/src/pop.cc gpc++0.5.2/src/pop.cc
--- gpc++0.5.2_oryg/src/pop.cc	1997-02-16 13:51:25.000000000 +0100
+++ gpc++0.5.2/src/pop.cc	2005-06-03 17:25:35.000000000 +0200
@@ -45,7 +45,7 @@
 // The Population class has most of the genetic programming operator
 // applied to it. 
 
-#include <iomanip.h>
+#include <iomanip>
 #include <stdio.h>
 
 #include "gp.h"
